stacks-signer/src/client/stacks_client.rs

```diff
@@ -67,6 +67,34 @@ impl From<&Config> for StacksClient {
 }
 
 impl StacksClient {
+    /// Retrieve the stacks tip consensus hash from the stacks node
+    pub fn get_stacks_tip_consensus_hash(&self) -> Result<String, ClientError> {
+        let send_request = || {
+            self.stacks_node_client
+                .get(self.core_info_path())
+                .send()
+                .map_err(backoff::Error::transient)
+        };
+
+        let response = retry_with_exponential_backoff(send_request)?;
+        if !response.status().is_success() {
+            return Err(ClientError::RequestFailure(response.status()));
+        }
+
+        let json_response = response
+            .json::<serde_json::Value>()
+            .map_err(ClientError::ReqwestError)?;
+
+        json_response["stacks_tip_consensus_hash"]
```

Comment:
I would deserialize the response into RPCPeerInfoData as the json response could be incorrect here and this hard array access would cause the signers to crash. It might be overkill to deserialize the entire response, but would at least convert the array access to a json_response.get("stacks_tip_consensus_hash") and map the result :)

Reply:
whoops. Updated to use the .get for now